fili-core/src/main/java/com/yahoo/bard/webservice/web/filterparser/ApiFiltersListListener.java

```diff
@@ -0,0 +1,219 @@
+                dimensionFiltersMap.put(dimension, new LinkedHashSet<>());
+            }
+            Set<ApiFilter> filterSet = dimensionFiltersMap.get(dimension);
+            filterSet.add(filter);
+            dimensionFiltersMap.put(dimension, filterSet);
+        }
+    }
+
+    @Override
+    public void exitFilterComponent(@NotNull FiltersParser.FilterComponentContext f) {
+        Dimension dimension = extractDimension(f);
+        if (dimension == null) {
+            return;
+        }
+
+        DimensionField dimensionField = extractField(f, dimension);
+        if (dimensionField == null) {
+            return;
+        }
+
+        FilterOperation operation = extractOperation(f);
+        if (operation == null) {
+            return;
+        }
+
+        Set<String> values = extractValues(f);
+
+        ApiFilter filter = new ApiFilter(dimension, dimensionField, operation, values);
+
+        filters.put(f, filter);
+    }
+
+    /**
+     * Extract values from the filter.
+     * @param f The filter context
+     * @return the list of values
+     */
+    protected Set<String> extractValues(FiltersParser.FilterComponentContext f) {
+        Set<String> values = new LinkedHashSet<>();
+        for (TerminalNode tok : f.values().VALUE()) {
+            String val = tok.getText();
+            values.add(val);
+        }
+        return values;
+    }
+
+    /**
+     * Extract the operation from the filter.
+     * @param f THe filter context
+     * @return the operation
+     */
+    protected FilterOperation extractOperation(FiltersParser.FilterComponentContext f) {
+        String operationName = f.OPERATOR().getText();
+        FilterOperation operation;
+        try {
+            operation = FilterOperation.valueOf(operationName);
+            if ((operation == FilterOperation.startswith || operation == FilterOperation.contains) &&
+                            !BardFeatureFlag.DATA_FILTER_SUBSTRING_OPERATIONS.isOn()) {
+                errors.add(new BadApiRequestException(
+                        ErrorMessageFormat.FILTER_SUBSTRING_OPERATIONS_DISABLED.format())
+                );
+            }
+
+        } catch (IllegalArgumentException ignored) {
+            LOG.debug(FILTER_OPERATOR_INVALID.logFormat(operationName));
+            errors.add(new BadFilterException(FILTER_OPERATOR_INVALID.format(operationName)));
+            return null;
+        }
+        return operation;
+    }
+
+    /**
+     * Extract the field portion of a filter.
+     * @param f The filter context
+     * @param dimension The dimension for the field
+     * @return the DimensionField
+     */
+    protected DimensionField extractField(FiltersParser.FilterComponentContext f, Dimension dimension) {
+        String dimensionFieldName;
+        dimensionFieldName = f.field().getText();
+        DimensionField dimensionField;
+        try {
+            dimensionField = dimension.getFieldByName(dimensionFieldName);
```

Comment:
You could probably just have a `return` statement here, and save yourself the extra variable.